test/unit/fields_definition_test.rb

```diff
@@ -38,262 +46,228 @@ class FieldsDefinitionTest < ActiveSupport::TestCase
     assert fd.visible_for?(provider_user)
   end
 
-  should 'not set unregistered target' do
+  test 'not set unregistered target' do
     fd = FieldsDefinition.new
     assert FieldsDefinition.targets.exclude?("Foo")
 
     fd.target = "Foo"
     assert fd.target.nil?
   end
 
-  context "validations" do
-    setup do
-      @provider = FactoryBot.create(:provider_account)
-      FieldsDefinition.where(:account_id => @provider.id).delete_all # removing default ones
-      @field_definition1 = FieldsDefinition.create(:account => @provider, :name => 'org_name', :target => 'Account', :label => 'foo')
-    end
+  test "not in database modificable fields not allow hidden AND required" do
+    fd = new_field_definition(hidden: true, required: true)
+    assert_not fd.valid?
+    fd.hidden = false
+    assert fd.valid?
+  end
+
+  test "not in database modificable fields not allow readonly AND required" do
+    fd = new_field_definition(read_only: true, required: true)
+    assert_not fd.valid?
+    fd.read_only = false
+    assert fd.valid?
+  end
+
+  test "not in database modificable fields allow choices property" do
+    a = new_field_definition(choices: %w[a b])
+    assert a.valid?
+  end
+end
 
-    should validate_presence_of :name
-    should validate_presence_of :label
-    should validate_presence_of :target
+class ValidationTest < ActiveSupport::TestCase
+  def setup
+    @provider = FactoryBot.create(:provider_account)
+    FieldsDefinition.where(account_id: @provider.id).delete_all # removing default ones
 
-    should "err on equal name for the same target and provider" do
-      assert_raise ActiveRecord::RecordInvalid do
-        FieldsDefinition.create!(:account => @provider, :name => 'org_name', :target => 'Account', :label => 'bar2')
-      end
+    @fd = new_field_definition(label: 'lal', target: 'User', name: 'first_name', choices: %w[a b])
+  end
+
+  test 'validate presence of' do
+    assert validate_presence_of :name
+    assert validate_presence_of :label
+    assert validate_presence_of :target
```

Comment:
where do these checks come from?

Reply:
From the gem `shoulda-matchers`

Reply:
Aren't we removing that? Or not?

Reply:
Only `shoulda-context` at the moment.

Reply:
Actually I'm not sure. Now I saw unstable:
https://app.circleci.com/pipelines/github/3scale/porta/19182/workflows/b9d5fa7b-eebb-4ba6-b839-171b8c555b19/jobs/229034/tests

Also I see a deprecation warning from `shoulda-matchers`:
```
~/.asdf/installs/ruby/2.6.7/lib/ruby/gems/2.6.0/gems/shoulda-matchers-2.8.0/lib/shoulda/matchers/active_model/validate_inclusion_of_matcher.rb:251: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
```

Investigating.

Reply:
Ok, the failures were not caused by shoulda [1] but by test order. The deprecation is an issue.

[1] https://github.com/3scale/porta/pull/2820/commits/ad30893024dc0fd2f0b5c7310566c4e13060d242